### PR TITLE
FusedOp class removal

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 Final FusedOp class removal:
- [x] Removal of fused_op.hpp and fused_op.cpp
- [x] Removal of fused_op.hpp includes
- [x] Removal of `supports_decompose` util function from  `ngraph/core/src/op/util/op_types.cpp`
- [x] Removal of deprecated `fused_op.in.cpp` tests file

### Tickets:
 - related to 30457

### CODE REVIEW

**Review by tomdol — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

File: ngraph/test/CMakeLists.txt [hunk 1]
```diff
@@ -448,7 +448,6 @@ set(MULTI_TEST_SRC
     backend/floor.in.cpp
     backend/floor_mod.in.cpp
     backend/function_name.in.cpp
-    backend/fused_op.in.cpp
```
Comment (ilyachur) [inline ngraph/test/CMakeLists.txt:451]: I think you also should delete `backend/fused_op.in.cpp` file

Comment (mitruska) [reply ngraph/test/CMakeLists.txt:451]: Ok, as the tests migration was a part of ops revision,
 just quick check to ensure all operations from `fused_op.in.cpp` have existing tests: 

HardSigmoid tests -> [ngraph/test/backend/hard_sigmoid.in.cpp](https://github.com/mitruska/openvino/blob/mitruska/fusedop_class_removal/ngraph/test/backend/hard_sigmoid.in.cpp)
MVN tests -> [docs/template_plugin/tests/functional/op_reference/mvn.cpp](https://github.com/mitruska/openvino/blob/mitruska/fusedop_class_removal/docs/template_plugin/tests/functional/op_reference/mvn.cpp)
SquaredDifference tests -> [ngraph/test/backend/squared_difference.in.cpp](https://github.com/mitruska/openvino/blob/mitruska/fusedop_class_removal/ngraph/test/backend/squared_difference.in.cpp)
SpaceToDepth tests ->[ngraph/test/backend/space_to_depth.in.cpp](https://github.com/mitruska/openvino/blob/mitruska/fusedop_class_removal/ngraph/test/backend/space_to_depth.in.cpp)

Seems can be removed.

Comment (ilyachur) [reply ngraph/test/CMakeLists.txt:451]: In any case you have removed this file from CMake script, it means that we exclude it from the build

Comment (mitruska) [reply ngraph/test/CMakeLists.txt:451]: Sure, done.

File: ngraph/core/src/op/util/op_types.cpp [hunk 2]
```diff
@@ -50,12 +49,6 @@ bool ngraph::op::supports_auto_broadcast(const ngraph::Node* node) {
            dynamic_cast<const ngraph::op::util::BinaryElementwiseArithmetic*>(node) != nullptr;
 }
 
-bool ngraph::op::supports_decompose(const ngraph::Node* node) {
```
[the remark this answers is not in the document]
Comment (mitruska) [reply ngraph/core/src/op/util/op_types.cpp:53]: You're right. Thank you. 
Now I'm wondering, is it okay to simply remove these methods or we should change them to return false and deprecate to remove later?

[the remark this answers is not in the document]
Comment (ilyachur) [reply ngraph/core/src/op/util/op_types.cpp:53]: I think we can remove these methods.

[the remark this answers is not in the document]
Comment (mitruska) [reply ngraph/core/src/op/util/op_types.cpp:53]: Done

**Review by ilyachur — APPROVED** [review]

**Pull request merged**